### PR TITLE
ROX-22354: autorotate DP secrets if data has changed

### PR DESCRIPTION
**Pull request by johannes94**
## Description

This PR changes the dataplane secret backup solution so that changes in secret data on a data plane cluster will automatically also updated the encrypted secret backup for quick recovery stored in FM's DB.

With this change the backup in FM will store changed secrets, even if they've been compromised and are invalid. This is ok, since the goal of this backup is quick recovery in case of accidental deletion or cluster failure. In case of compromised/invalid secrets for any reason we can use the S3 velero backups done by OSD.

Runbook Change:
https://gitlab.cee.redhat.com/stackrox/acs-cloud-service/runbooks/-/merge_requests/112

## Checklist (Definition of Done)

- [x] Unit and integration tests added
- [x] Added test description under `Test manual`
- [x] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

e2e tests in CI tests that backup / restore of secrets is still working like expected
Added additional unit testing to make sure equality logic for sums is correct.

### CODE REVIEW

File: internal/dinosaur/pkg/migrations/20240703120000_add_secret_data_sum_to_central_request.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/pkg/errors"
+	"github.com/stackrox/acs-fleet-manager/pkg/db"
+	"gorm.io/gorm"
+)
+
+func addSecretDataSumToCentralRequest() *gormigrate.Migration {
+	type CentralRequest struct {
+		db.Model
+		SecretDataSum string `json:"secret_data_sum"`
```
Comment (ludydoo) [inline internal/dinosaur/pkg/migrations/20240703120000_add_secret_data_sum_to_central_request.go:18]: Can we add the algorithm in the field name? e.g. `secret_data_sha256_sum`

File: openapi/fleet-manager-private.yaml [hunk 2]
```diff
@@ -459,6 +461,9 @@ components:
           type: object
           additionalProperties:
             type: string
+        secretDataSum:
```
Comment (ludydoo) [inline openapi/fleet-manager-private.yaml:464]: Can we also add the `sha256SecretDataSum` here ?

File: fleetshard/pkg/central/reconciler/reconciler.go [hunk 3]
```diff
@@ -875,18 +881,18 @@ func (r *CentralReconciler) collectSecrets(ctx context.Context, remoteCentral *p
 	return secrets, nil
 }
 
-func (r *CentralReconciler) collectSecretsEncrypted(ctx context.Context, remoteCentral *private.ManagedCentral) (map[string]string, error) {
+func (r *CentralReconciler) collectSecretsEncrypted(ctx context.Context, remoteCentral *private.ManagedCentral) (map[string]string, string, error) {
```
[the remark this answers is not in the document]
Comment (johannes94) [reply fleetshard/pkg/central/reconciler/reconciler.go]: Good idea 👍 Did that

File: internal/dinosaur/pkg/migrations/20240703120000_add_secret_data_sum_to_central_request.go (hunk 1 above)
Comment (johannes94) [reply internal/dinosaur/pkg/migrations/20240703120000_add_secret_data_sum_to_central_request.go]: Changed all occurrences to have the algorithm in the field name.

**Review by ludydoo — APPROVED** [review]

**Pull request merged**